spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServicePropertySourceLocator.java

```diff
@@ -92,6 +92,9 @@ private List<String> combineProfiles(ConfigClientProperties properties,
 		if (!ObjectUtils.isEmpty(properties.getProfile())) {
 			combinedProfiles = Stream.of(properties.getProfile().split(",")).map(String::trim).filter(s -> !s.isEmpty())
 					.collect(Collectors.toList());
+			// If spring.cloud.config.profiles is set it should override all other
+			// profiles set other ways
+			return combinedProfiles;
```

Comment:
This makes this method weird right? It's no longer combining profiles. It's returning either the config props profiles, or the active profiles, or default profiles